src/pages/studyView/table/FixedHeaderTable.tsx

```diff
@@ -0,0 +1,153 @@
+import * as React from "react";
+import {Column, LazyMobXTableStore, SortDirection} from "../../../shared/components/lazyMobXTable/LazyMobXTable";
+import {
+    Column as RVColumn,
+    SortDirection as RVSortDirection,
+    Table as RVTable,
+    TableCellProps
+} from 'react-virtualized';
+import 'react-virtualized/styles.css';
+import {action, observable} from "mobx";
+import styles from "./tables.module.scss";
+import * as _ from 'lodash';
+import {observer} from "mobx-react";
+import classnames from 'classnames';
+import {If} from 'react-if';
+import {bind} from "bind-decorator";
+import {inputBoxChangeTimeoutEvent} from "../../../shared/lib/EventUtils";
+
+export type IFixedHeaderTableProps<T> = {
+    columns: Column<T>[],
+    data: T[];
+    sortBy?: string;
+    width?: number;
+    height?: number;
+    // This property is just used to force to rerender the table when selectedGenes is modified.
+    // We don't actually use the property anywhere in the table.
+    selectedGenes?: string[];
+    selectedRows?: number[];
+    showSelectSamples?: boolean;
+    afterSelectingRows?: () => void;
+};
+
+const RVSDTtoStrType = {
+    ['desc' as SortDirection]: RVSortDirection.DESC,
+    ['asc' as SortDirection]: RVSortDirection.ASC
+};
+
+@observer
+export default class FixedHeaderTable<T> extends React.Component<IFixedHeaderTableProps<T>, {}> {
+    private _store: LazyMobXTableStore<T>;
+    @observable private _sortBy: string;
+    @observable private _sortDirection: SortDirection;
+
+    public static defaultProps = {
+        showSelectSamples: false,
+        selectedRows: [],
+        selectedGenes: [],
+        width : 398,
+        height: 350,
+        sortBy: ''
+    };
+
+    constructor(props: IFixedHeaderTableProps<T>) {
+        super(props);
+        this._sortBy = props.sortBy!;
+        const sortByColumn = _.find(this.props.columns, column => column.name === this._sortBy);
+        this._sortDirection = _.isUndefined(sortByColumn) ? 'desc' as 'desc' : sortByColumn.defaultSortDirection || 'desc' as 'desc';
+
+        this.initDataStore();
+    }
+
+    @action
+    initDataStore() {
+        this._store = new LazyMobXTableStore<T>({
+            columns: this.props.columns,
+            data: this.props.data,
+            initialSortColumn: this._sortBy,
+            initialSortDirection: this._sortDirection
+        });
+    }
+
+    @bind
+    rowClassName({index}: any) {
+        if (_.includes(this.props.selectedRows, index)) {
```

Comment:
changing sorting order highlights different records as the index of the selected record changes